common/js/solid.js

```diff
@@ -0,0 +1,440 @@
+/* global owaspPasswordStrengthTest, TextEncoder, crypto */
+(function () {
+  'use strict'
+
+  var PasswordValidator = function (passwordField, repeatedPasswordField) {
+    if (
+      passwordField === null || passwordField === undefined ||
+      repeatedPasswordField === null || repeatedPasswordField === undefined
+    ) {
+      return
+    }
+
+    this.passwordField = passwordField
+    this.repeatedPasswordField = repeatedPasswordField
+
+    this.fetchDomNodes()
+    this.bindEvents()
+
+    this.currentStrengthLevel = 0
+    this.errors = []
+  }
+
+  PasswordValidator.prototype.FEEDBACK_SUCCESS = 'success'
+  PasswordValidator.prototype.FEEDBACK_WARNING = 'warning'
+  PasswordValidator.prototype.FEEDBACK_ERROR = 'error'
+
+  PasswordValidator.prototype.ICON_SUCCESS = 'glyphicon-ok'
+  PasswordValidator.prototype.ICON_WARNING = 'glyphicon-warning-sign'
+  PasswordValidator.prototype.ICON_ERROR = 'glyphicon-remove'
+
+  PasswordValidator.prototype.VALIDATION_SUCCESS = 'has-success'
+  PasswordValidator.prototype.VALIDATION_WARNING = 'has-warning'
+  PasswordValidator.prototype.VALIDATION_ERROR = 'has-error'
+
+  PasswordValidator.prototype.STRENGTH_PROGRESS_0 = 'progress-bar-danger level-0'
+  PasswordValidator.prototype.STRENGTH_PROGRESS_1 = 'progress-bar-danger level-1'
+  PasswordValidator.prototype.STRENGTH_PROGRESS_2 = 'progress-bar-warning level-2'
+  PasswordValidator.prototype.STRENGTH_PROGRESS_3 = 'progress-bar-success level-3'
+  PasswordValidator.prototype.STRENGTH_PROGRESS_4 = 'progress-bar-success level-4'
```

Comment:
Lett's just have the above as `const`, not on the prototype.

Reply:
fixed